prow/plugins/projectqueries/projectqueries.go

```diff
@@ -0,0 +1,208 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package projectqueries is a plugin to auto add pull requests to project boards based on a search query
+package projectqueries
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	pluginName = "project-queries"
+)
+
+// TODO Create a new handler for issues, look in hook/server.go
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The projects plugin auto adds issues/PRs to projects based on a search query",
+		Config: func(config *plugins.Configuration) map[string]string {
+			configMap := make(map[string]string)
+			configString := ""
+			for _, projectQuery := range config.ProjectQueries {
+				projectString := fmt.Sprintf("Project Org: \"%s\", Project Name: \"%s\", Project Column: \"%s\"", projectQuery.Org, projectQuery.Name, projectQuery.Column)
+				labelsString := ""
+				for _, label := range projectQuery.Query.Labels {
+					labelsString = fmt.Sprintf("%s label:%s", labelsString, label)
+				}
+				queryString := fmt.Sprintf("Query: \"org:%s state:%s%s\"", projectQuery.Query.Org, projectQuery.Query.State, labelsString)
+				configString = fmt.Sprintf("%s\n%s %s", configString, projectString, queryString)
+			}
+			configMap[""] = configString
+			return configMap
+		}(config),
+	}
+	return pluginHelp, nil
+}
+
+func handlePullRequest(pc plugins.Agent, pe github.PullRequestEvent) error {
+	return handlePR(pc.GitHubClient, pc.PluginConfig.ProjectQueries, pc.Logger, pe)
+}
+
+// Strict subset of *github.Client methods.
+type githubClient interface {
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	GetOrgProjects(org string) ([]github.Project, error)
+	GetProjectColumns(projectID int) ([]github.ProjectColumn, error)
+	CreateProjectCard(columnID int, projectCard github.ProjectCard) (*github.ProjectCard, error)
+}
+
+func handlePR(gc githubClient, projectQueries []plugins.ProjectQuery, le *logrus.Entry, pe github.PullRequestEvent) error {
+
+	if len(projectQueries) == 0 {
+		return nil
+	}
+
+	// Only respond to label add or issue/PR open events
+	if !isPRChanged(pe) {
+		return nil
+	}
+
+	// Get any ProjectQueries that match this PR
+	matchedProjectQueries, err := getMatchingProjectQueries(gc, projectQueries, pe)
+	if err != nil {
+		return err
+	}
+	if len(matchedProjectQueries) == 0 {
```

Comment:
You'll get this behavior by default with the `for _, matchedProjectQuery := ` loop below if the slice is empty, so you don't need this block

Reply:
I've removed the redundant block.